src/methods/iterator.js

```diff
@@ -177,12 +269,18 @@ export function IteratorClose(realm: Realm, iterator: ObjectValue, completion: A
   if (ret instanceof UndefinedValue) return completion;
 
   // 5. Let innerResult be Call(return, iterator, « »).
-  let innerResult = Call(realm, ret.throwIfNotConcrete(), iterator, []);
+  let innerResult;
+  try {
+    innerResult = Call(realm, ret.throwIfNotConcrete(), iterator, []);
+  } catch (error) {
+    innerResult = error;
+  }
 
   // 6. If completion.[[Type]] is throw, return Completion(completion).
   if (completion instanceof ThrowCompletion) return completion;
 
   // 7. If innerResult.[[Type]] is throw, return Completion(innerResult).
+  if (innerResult instanceof ThrowCompletion) return innerResult;
 
   // 8. If Type(innerResult.[[Value]]) is not Object, throw a TypeError exception.
   if (!(innerResult instanceof ObjectValue)) {
```

Comment:
I guess the spec language strongly suggests throwing this, but it is still a wee bit strange to me.